src/remote/index.js

```diff
@@ -165,17 +165,34 @@ export default class Remote implements Side {
     }
   }
 
+  async moveFileAsync (newMetadata: Metadata, oldMetadata: Metadata): Promise<Metadata> {
+    // TODO: v3: Call addFile() when !from.remote
+    // TODO: v3: Call addFile() when file not found on cozy
+    const [newDirPath, newName]: [string, string] = conversion.extractDirAndName(newMetadata.path)
+    const newDir: RemoteDoc = await this.remoteCozy.findDirectoryByPath(newDirPath)
+    const newRemoteDoc: RemoteDoc = await this.remoteCozy.updateAttributesById(oldMetadata.remote._id, {
+      name: newName,
+      dir_id: newDir._id
+    })
+
+    return conversion.createMetadata(newRemoteDoc)
+  }
+
+  moveFile (doc: Metadata, from: Metadata, callback: Callback): void {
+    try {
+      this.moveFileAsync(doc, from).asCallback(callback)
+    } catch (err) {
```

Comment:
I'm not convinced that the `try/catch` here is useful (we are not in an async function)